tests/test_profiling_result_printer.py

```diff
@@ -0,0 +1,163 @@
+from typing import Any
+
+import pytest
+import numpy
+
+import modmesh
+from modmesh.profiling import (
+    ProfilingColumnData,
+    ProfilingTableBuilder,
+    ProfilingResultPrinter,
+)
+
+
+class TestProfilingResultPrinter:
+    def run_profile_function(self):
+        _ = modmesh.CallProfilerProbe("numpy_arange_100")
+        numpy.arange(0, 100, dtype="uint32")
+
+    @pytest.fixture
+    def profiling_result_fixture(self) -> dict[str, Any]:
+        modmesh.call_profiler.reset()
+
+        self.run_profile_function()
+
+        result: dict[str, Any] = modmesh.call_profiler.result()["children"]
+        return result
+
+    def test_printer_getitem_with_valid_key_should_return_correct_data(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
+            profiling_result_fixture
+        )
+
+        assert printer["numpy_arange_100"].name == "numpy_arange_100"
+
+    def test_printer_getitem_with_absent_key_should_raise_value_error(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
+            profiling_result_fixture
+        )
+
+        try:
+            printer["numpy_arange_1000"]
+            assert False
+        except ValueError:
+            assert True
+
+    def test_printer_add_column_should_have_correct_column(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
+            profiling_result_fixture
+        )
+
+        tot: float = printer["numpy_arange_100"].total_time
+        printer.add_column("tot_scale_10", lambda r: r.total_time * 10)
+
+        col = list(
+            filter(
+                lambda cols: cols.column_name == "tot_scale_10",
+                printer.column_datas,
+            )
+        )[0]
+        assert col.column_data[0] == tot * 10
+
+    def test_printer_constructor_should_have_func_name_column_as_default(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
```

Comment:
You should also test the default columns with null input to the constructor.

Reply:
OK. I add test for testing null value in constructor and handled in my profiling table.